### PR TITLE
port examples to new api

### PR DESCRIPTION
**Pull request by hugomrdias**

### CODE REVIEW

File: examples/bundle-webpack/src/App.js [hunk 1]
```diff
@@ -16,24 +16,27 @@ class App extends React.Component {
...
+    this.setState({
+      id: id.id,
+      version: id.agentVersion,
+      protocol_version: id.protocolVersion
     })
-    ipfs.add([Buffer.from(stringToUse)], (err, res) => {
-      if (err) throw err
-      const hash = res[0].hash
+
+    const source = await ipfs.add(stringToUse)
```
Comment (alanshaw) [inline examples/bundle-webpack/src/App.js]:
```suggestion
    const source = ipfs.add(stringToUse)
```

File: examples/bundle-webpack/src/App.js [hunk 2]
```diff
@@ -16,24 +16,27 @@ class App extends React.Component {
...
+      console.log("TCL: App -> forawait -> file", file)
+      const hash = file.path
       this.setState({ added_file_hash: hash })
-      ipfs.cat(hash, (err, data) => {
-        if (err) throw err
-        this.setState({ added_file_contents: data.toString() })
-      })
-    })
+
+      const source = await ipfs.cat(hash)
```
Comment (alanshaw) [inline examples/bundle-webpack/src/App.js]:
```suggestion
      const source = ipfs.cat(hash)
```

File: examples/files-api/files-api.js [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
-    res.pipe(process.stdout)
-  } else {
-    console.log(res)
+const run = async () => {
+  await ipfs.files.write(
+    '/temp/hello-world',
+    Buffer.from('Hello, world!'),
+    { create: true, parents: true }
+  )
+  const source = await ipfs.files.ls('/temp')
```
Comment (alanshaw) [inline examples/files-api/files-api.js]:
```suggestion
  const source = ipfs.files.ls('/temp')
```

File: examples/upload-file-via-browser/src/App.js [hunk 4]
```diff
@@ -28,24 +29,27 @@ class App extends React.Component {
...
-      .then((response) => {
-        console.log(response)
-        ipfsId = response[0].hash
-        console.log(ipfsId)
-        this.setState({ added_file_hash: ipfsId })
-      }).catch((err) => {
-        console.error(err)
-      })
+  async saveToIpfs (files) {
+    const source = await this.ipfs.add(
```
Comment (alanshaw) [inline examples/upload-file-via-browser/src/App.js]:
```suggestion
    const source = this.ipfs.add(
```

File: examples/upload-file-via-browser/src/App.js [hunk 5]
```diff
@@ -54,16 +58,16 @@ class App extends React.Component {
...
-        console.log(response)
-        // CID of wrapping directory is returned last
-        ipfsId = response[response.length - 1].hash
-        console.log(ipfsId)
-        this.setState({ added_file_hash: ipfsId })
-      }).catch((err) => {
-        console.error(err)
-      })
+
+    const source = await this.ipfs.add(fileDetails, options)
```
Comment (alanshaw) [inline examples/upload-file-via-browser/src/App.js]:
```suggestion
    const source = this.ipfs.add(fileDetails, options)
```

**Comment by alanshaw** [comment]
@hugomrdias would you mind rebasing this so I can look at just your changes?

**Comment by hugomrdias** [comment]
done

**Comment by hugomrdias** [comment]
browserify doesn't support deps that use esm.

**Comment by achingbrain** [comment]
Could we use something like this to support esm in browserify? https://github.com/mattdesl/esmify

**Comment by mikeal** [comment]
https://github.com/browserify/browserify/issues/1186

**Comment by alanshaw** [comment]
I'm going to merge this so that we have up to date examples for the official release and open an issue so that we can get a browserify example re-added.

**Pull request merged**